> CI failed https://github.com/RediSearch/RediSearch/actions/runs/17573765484/job/49914586307
> 
> ```
> error: linking with `cc` failed: exit status: 1
>   |
>   = note:  "cc" "/var/folders/vk/nx37ffx50hv5djclhltc26vw0000gn/T/rustcGZ67Td/symbols.o" "<1 object files omitted>" "-linverted_index" "-lbuffer" "<sysroot>/lib/rustlib/x86_64-apple-darwin/lib/{libcompiler_builtins-*}.rlib" "-liconv" "-lSystem" "-lc" "-lm" "-arch" "x86_64" "-mmacosx-version-min=10.12.0" "-L" "/Users/runner/work/RediSearch/RediSearch/bin/macos-x64-release/search-community/src/inverted_index" "-L" "/Users/runner/work/RediSearch/RediSearch/bin/macos-x64-release/search-community/src/buffer" "-o" "/Users/runner/work/RediSearch/RediSearch/src/redisearch_rs/../../bin/redisearch_rs/optimised_test/deps/inverted_index_bencher-dbb5f02fc569baac" "-Wl,-dead_strip" "-nodefaultlibs"
>   = note: some arguments are omitted. use `--verbose` to show all linker arguments
>   = note: ld: warning: object file (/Users/runner/work/RediSearch/RediSearch/bin/macos-x64-release/search-community/src/inverted_index/libinverted_index.a[2](inverted_index.c.o)) was built for newer 'macOS' version (13.7) than being linked (10.12)
>           ld: warning: object file (/Users/runner/work/RediSearch/RediSearch/bin/macos-x64-release/search-community/src/buffer/libbuffer.a[2](buffer.c.o)) was built for newer 'macOS' version (13.7) than being linked (10.12)
>           ld: Undefined symbols:
>             _RSGlobalConfig, referenced from:
>                 _encodeNumeric in libinverted_index.a[2](inverted_index.c.o)
>             _isWithinRadius, referenced from:
>                 _readNumeric in libinverted_index.a[2](inverted_index.c.o)
>           clang: error: linker command failed with exit code 1 (use -v to see invocation)
> ```
> 
> @GlenDC : I think you are using MacOS, right? Can you reproduce this problem with this branch?

Nope. But on this branch I have other error on MacOS
(FYI I can test on MacOS 15 arm64, Windows 11 amd64 and Ubuntu 20.25 amd64))

```
error[E0412]: cannot find type `IndexDecoderCtx` in this scope
    --> /Users/glendc/code/github.com/redisearch/RediSearch/src/redisearch_rs/../../bin/redisearch_rs/optimised_test/build/inverted_index_bencher-3dd870ec90f6ca32/out/bindings.rs:1904:22
     |
1904 |         arg2: *const IndexDecoderCtx,
     |                      ^^^^^^^^^^^^^^^ not found in this scope
     |
help: consider importing this union
    -->  inverted_index_bencher/src/ffi.rs:25:5
     |
25   +     use ffi::IndexDecoderCtx;
     |

error[E0412]: cannot find type `IndexDecoderCtx` in this scope
    --> /Users/glendc/code/github.com/redisearch/RediSearch/src/redisearch_rs/../../bin/redisearch_rs/optimised_test/build/inverted_index_bencher-3dd870ec90f6ca32/out/bindings.rs:1912:22
     |
1901 | / pub type IndexDecoder = ::std::option::Option<
1902 | |     unsafe extern "C" fn(
1903 | |         arg1: *mut IndexBlockReader,
1904 | |         arg2: *const IndexDecoderCtx,
1905 | |         out: *mut RSIndexResult,
1906 | |     ) -> bool,
1907 | | >;
     | |__- similarly named type alias `IndexDecoder` defined here
...
1912 |           arg2: *const IndexDecoderCtx,
     |                        ^^^^^^^^^^^^^^^
     |
help: a type alias with a similar name exists
     |
1912 -         arg2: *const IndexDecoderCtx,
1912 +         arg2: *const IndexDecoder,
     |
help: consider importing this union
    -->  inverted_index_bencher/src/ffi.rs:25:5
     |
25   +     use ffi::IndexDecoderCtx;
     |

error[E0412]: cannot find type `IndexDecoderCtx` in this scope
    --> /Users/glendc/code/github.com/redisearch/RediSearch/src/redisearch_rs/../../bin/redisearch_rs/optimised_test/build/inverted_index_bencher-3dd870ec90f6ca32/out/bindings.rs:1938:21
     |
1901 | / pub type IndexDecoder = ::std::option::Option<
1902 | |     unsafe extern "C" fn(
1903 | |         arg1: *mut IndexBlockReader,
1904 | |         arg2: *const IndexDecoderCtx,
1905 | |         out: *mut RSIndexResult,
1906 | |     ) -> bool,
1907 | | >;
     | |__- similarly named type alias `IndexDecoder` defined here
...
1938 |       pub decoderCtx: IndexDecoderCtx,
     |                       ^^^^^^^^^^^^^^^
     |
help: a type alias with a similar name exists
     |
1938 -     pub decoderCtx: IndexDecoderCtx,
1938 +     pub decoderCtx: IndexDecoder,
     |
help: consider importing this union
    -->  inverted_index_bencher/src/ffi.rs:25:5
     |
25   +     use ffi::IndexDecoderCtx;
     |

error[E0412]: cannot find type `IndexDecoderCtx` in this scope
    --> /Users/glendc/code/github.com/redisearch/RediSearch/src/redisearch_rs/../../bin/redisearch_rs/optimised_test/build/inverted_index_bencher-3dd870ec90f6ca32/out/bindings.rs:1959:64
     |
1901 | / pub type IndexDecoder = ::std::option::Option<
1902 | |     unsafe extern "C" fn(
1903 | |         arg1: *mut IndexBlockReader,
1904 | |         arg2: *const IndexDecoderCtx,
1905 | |         out: *mut RSIndexResult,
1906 | |     ) -> bool,
1907 | | >;
     | |__- similarly named type alias `IndexDecoder` defined here
...
1959 |       pub fn NewIndexReader(idx: *const InvertedIndex, ctx: *mut IndexDecoderCtx)
     |                                                                  ^^^^^^^^^^^^^^^
     |
help: a type alias with a similar name exists
     |
1959 -     pub fn NewIndexReader(idx: *const InvertedIndex, ctx: *mut IndexDecoderCtx)
1959 +     pub fn NewIndexReader(idx: *const InvertedIndex, ctx: *mut IndexDecoder)
     |
help: consider importing this union
    -->  inverted_index_bencher/src/ffi.rs:25:5
     |
25   +     use ffi::IndexDecoderCtx;
     |

error[E0412]: cannot find type `IndexDecoderCtx` in this scope
    --> /Users/glendc/code/github.com/redisearch/RediSearch/src/redisearch_rs/../../bin/redisearch_rs/optimised_test/build/inverted_index_bencher-3dd870ec90f6ca32/out/bindings.rs:2018:50
     |
1901 | / pub type IndexDecoder = ::std::option::Option<
1902 | |     unsafe extern "C" fn(
1903 | |         arg1: *mut IndexBlockReader,
1904 | |         arg2: *const IndexDecoderCtx,
1905 | |         out: *mut RSIndexResult,
1906 | |     ) -> bool,
1907 | | >;
     | |__- similarly named type alias `IndexDecoder` defined here
...
2018 |       pub fn NewIndexDecoderCtx_NumericFilter() -> IndexDecoderCtx;
     |                                                    ^^^^^^^^^^^^^^^
     |
help: a type alias with a similar name exists
     |
2018 -     pub fn NewIndexDecoderCtx_NumericFilter() -> IndexDecoderCtx;
2018 +     pub fn NewIndexDecoderCtx_NumericFilter() -> IndexDecoder;
     |
help: consider importing this union
    -->  inverted_index_bencher/src/ffi.rs:25:5
     |
25   +     use ffi::IndexDecoderCtx;
     |

error[E0412]: cannot find type `IndexDecoderCtx` in this scope
    --> /Users/glendc/code/github.com/redisearch/RediSearch/src/redisearch_rs/../../bin/redisearch_rs/optimised_test/build/inverted_index_bencher-3dd870ec90f6ca32/out/bindings.rs:2021:56
     |
1901 | / pub type IndexDecoder = ::std::option::Option<
1902 | |     unsafe extern "C" fn(
1903 | |         arg1: *mut IndexBlockReader,
1904 | |         arg2: *const IndexDecoderCtx,
1905 | |         out: *mut RSIndexResult,
1906 | |     ) -> bool,
1907 | | >;
     | |__- similarly named type alias `IndexDecoder` defined here
...
2021 |       pub fn NewIndexDecoderCtx_MaskFilter(mask: u32) -> IndexDecoderCtx;
     |                                                          ^^^^^^^^^^^^^^^
     |
help: a type alias with a similar name exists
     |
2021 -     pub fn NewIndexDecoderCtx_MaskFilter(mask: u32) -> IndexDecoderCtx;
2021 +     pub fn NewIndexDecoderCtx_MaskFilter(mask: u32) -> IndexDecoder;
     |
help: consider importing this union
    -->  inverted_index_bencher/src/ffi.rs:25:5
     |
25   +     use ffi::IndexDecoderCtx;
     |

error[E0412]: cannot find type `IndexDecoderCtx` in this scope
    --> /Users/glendc/code/github.com/redisearch/RediSearch/src/redisearch_rs/../../bin/redisearch_rs/optimised_test/build/inverted_index_bencher-3dd870ec90f6ca32/out/bindings.rs:2116:21
     |
1901 | / pub type IndexDecoder = ::std::option::Option<
1902 | |     unsafe extern "C" fn(
1903 | |         arg1: *mut IndexBlockReader,
1904 | |         arg2: *const IndexDecoderCtx,
1905 | |         out: *mut RSIndexResult,
1906 | |     ) -> bool,
1907 | | >;
     | |__- similarly named type alias `IndexDecoder` defined here
...
2116 |           ctx: *const IndexDecoderCtx,
     |                       ^^^^^^^^^^^^^^^
     |
help: a type alias with a similar name exists
     |
2116 -         ctx: *const IndexDecoderCtx,
2116 +         ctx: *const IndexDecoder,
     |
help: consider importing this union
    -->  inverted_index_bencher/src/ffi.rs:25:5
     |
25   +     use ffi::IndexDecoderCtx;
     |

error[E0412]: cannot find type `IndexDecoderCtx` in this scope
    --> /Users/glendc/code/github.com/redisearch/RediSearch/src/redisearch_rs/../../bin/redisearch_rs/optimised_test/build/inverted_index_bencher-3dd870ec90f6ca32/out/bindings.rs:2123:21
     |
1901 | / pub type IndexDecoder = ::std::option::Option<
1902 | |     unsafe extern "C" fn(
1903 | |         arg1: *mut IndexBlockReader,
1904 | |         arg2: *const IndexDecoderCtx,
1905 | |         out: *mut RSIndexResult,
1906 | |     ) -> bool,
1907 | | >;
     | |__- similarly named type alias `IndexDecoder` defined here
...
2123 |           ctx: *const IndexDecoderCtx,
     |                       ^^^^^^^^^^^^^^^
     |
help: a type alias with a similar name exists
     |
2123 -         ctx: *const IndexDecoderCtx,
2123 +         ctx: *const IndexDecoder,
     |
help: consider importing this union
    -->  inverted_index_bencher/src/ffi.rs:25:5
     |
25   +     use ffi::IndexDecoderCtx;
     |

error[E0412]: cannot find type `IndexDecoderCtx` in this scope
    --> /Users/glendc/code/github.com/redisearch/RediSearch/src/redisearch_rs/../../bin/redisearch_rs/optimised_test/build/inverted_index_bencher-3dd870ec90f6ca32/out/bindings.rs:2130:21
     |
1901 | / pub type IndexDecoder = ::std::option::Option<
1902 | |     unsafe extern "C" fn(
1903 | |         arg1: *mut IndexBlockReader,
1904 | |         arg2: *const IndexDecoderCtx,
1905 | |         out: *mut RSIndexResult,
1906 | |     ) -> bool,
1907 | | >;
     | |__- similarly named type alias `IndexDecoder` defined here
...
2130 |           ctx: *const IndexDecoderCtx,
     |                       ^^^^^^^^^^^^^^^
     |
help: a type alias with a similar name exists
     |
2130 -         ctx: *const IndexDecoderCtx,
2130 +         ctx: *const IndexDecoder,
     |
help: consider importing this union
    -->  inverted_index_bencher/src/ffi.rs:25:5
     |
25   +     use ffi::IndexDecoderCtx;
     |

error[E0412]: cannot find type `IndexDecoderCtx` in this scope
    --> /Users/glendc/code/github.com/redisearch/RediSearch/src/redisearch_rs/../../bin/redisearch_rs/optimised_test/build/inverted_index_bencher-3dd870ec90f6ca32/out/bindings.rs:2137:21
     |
1901 | / pub type IndexDecoder = ::std::option::Option<
1902 | |     unsafe extern "C" fn(
1903 | |         arg1: *mut IndexBlockReader,
1904 | |         arg2: *const IndexDecoderCtx,
1905 | |         out: *mut RSIndexResult,
1906 | |     ) -> bool,
1907 | | >;
     | |__- similarly named type alias `IndexDecoder` defined here
...
2137 |           ctx: *const IndexDecoderCtx,
     |                       ^^^^^^^^^^^^^^^
     |
help: a type alias with a similar name exists
     |
2137 -         ctx: *const IndexDecoderCtx,
2137 +         ctx: *const IndexDecoder,
     |
help: consider importing this union
    -->  inverted_index_bencher/src/ffi.rs:25:5
     |
25   +     use ffi::IndexDecoderCtx;
     |

error[E0412]: cannot find type `IndexDecoderCtx` in this scope
    --> /Users/glendc/code/github.com/redisearch/RediSearch/src/redisearch_rs/../../bin/redisearch_rs/optimised_test/build/inverted_index_bencher-3dd870ec90f6ca32/out/bindings.rs:2144:21
     |
1901 | / pub type IndexDecoder = ::std::option::Option<
1902 | |     unsafe extern "C" fn(
1903 | |         arg1: *mut IndexBlockReader,
1904 | |         arg2: *const IndexDecoderCtx,
1905 | |         out: *mut RSIndexResult,
1906 | |     ) -> bool,
1907 | | >;
     | |__- similarly named type alias `IndexDecoder` defined here
...
2144 |           ctx: *const IndexDecoderCtx,
     |                       ^^^^^^^^^^^^^^^
     |
help: a type alias with a similar name exists
     |
2144 -         ctx: *const IndexDecoderCtx,
2144 +         ctx: *const IndexDecoder,
     |
help: consider importing this union
    -->  inverted_index_bencher/src/ffi.rs:25:5
     |
25   +     use ffi::IndexDecoderCtx;
     |

error[E0412]: cannot find type `IndexDecoderCtx` in this scope
    --> /Users/glendc/code/github.com/redisearch/RediSearch/src/redisearch_rs/../../bin/redisearch_rs/optimised_test/build/inverted_index_bencher-3dd870ec90f6ca32/out/bindings.rs:2151:21
     |
1901 | / pub type IndexDecoder = ::std::option::Option<
1902 | |     unsafe extern "C" fn(
1903 | |         arg1: *mut IndexBlockReader,
1904 | |         arg2: *const IndexDecoderCtx,
1905 | |         out: *mut RSIndexResult,
1906 | |     ) -> bool,
1907 | | >;
     | |__- similarly named type alias `IndexDecoder` defined here
...
2151 |           ctx: *const IndexDecoderCtx,
     |                       ^^^^^^^^^^^^^^^
     |
help: a type alias with a similar name exists
     |
2151 -         ctx: *const IndexDecoderCtx,
2151 +         ctx: *const IndexDecoder,
     |
help: consider importing this union
    -->  inverted_index_bencher/src/ffi.rs:25:5
     |
25   +     use ffi::IndexDecoderCtx;
     |

error[E0412]: cannot find type `IndexDecoderCtx` in this scope
    --> /Users/glendc/code/github.com/redisearch/RediSearch/src/redisearch_rs/../../bin/redisearch_rs/optimised_test/build/inverted_index_bencher-3dd870ec90f6ca32/out/bindings.rs:2158:21
     |
1901 | / pub type IndexDecoder = ::std::option::Option<
1902 | |     unsafe extern "C" fn(
1903 | |         arg1: *mut IndexBlockReader,
1904 | |         arg2: *const IndexDecoderCtx,
1905 | |         out: *mut RSIndexResult,
1906 | |     ) -> bool,
1907 | | >;
     | |__- similarly named type alias `IndexDecoder` defined here
...
2158 |           ctx: *const IndexDecoderCtx,
     |                       ^^^^^^^^^^^^^^^
     |
help: a type alias with a similar name exists
     |
2158 -         ctx: *const IndexDecoderCtx,
2158 +         ctx: *const IndexDecoder,
     |
help: consider importing this union
    -->  inverted_index_bencher/src/ffi.rs:25:5
     |
25   +     use ffi::IndexDecoderCtx;
     |

error[E0412]: cannot find type `IndexDecoderCtx` in this scope
    --> /Users/glendc/code/github.com/redisearch/RediSearch/src/redisearch_rs/../../bin/redisearch_rs/optimised_test/build/inverted_index_bencher-3dd870ec90f6ca32/out/bindings.rs:2165:21
     |
1901 | / pub type IndexDecoder = ::std::option::Option<
1902 | |     unsafe extern "C" fn(
1903 | |         arg1: *mut IndexBlockReader,
1904 | |         arg2: *const IndexDecoderCtx,
1905 | |         out: *mut RSIndexResult,
1906 | |     ) -> bool,
1907 | | >;
     | |__- similarly named type alias `IndexDecoder` defined here
...
2165 |           ctx: *const IndexDecoderCtx,
     |                       ^^^^^^^^^^^^^^^
     |
help: a type alias with a similar name exists
     |
2165 -         ctx: *const IndexDecoderCtx,
2165 +         ctx: *const IndexDecoder,
     |
help: consider importing this union
    -->  inverted_index_bencher/src/ffi.rs:25:5
     |
25   +     use ffi::IndexDecoderCtx;
     |

error[E0412]: cannot find type `IndexDecoderCtx` in this scope
    --> /Users/glendc/code/github.com/redisearch/RediSearch/src/redisearch_rs/../../bin/redisearch_rs/optimised_test/build/inverted_index_bencher-3dd870ec90f6ca32/out/bindings.rs:2172:21
     |
1901 | / pub type IndexDecoder = ::std::option::Option<
1902 | |     unsafe extern "C" fn(
1903 | |         arg1: *mut IndexBlockReader,
1904 | |         arg2: *const IndexDecoderCtx,
1905 | |         out: *mut RSIndexResult,
1906 | |     ) -> bool,
1907 | | >;
     | |__- similarly named type alias `IndexDecoder` defined here
...
2172 |           ctx: *const IndexDecoderCtx,
     |                       ^^^^^^^^^^^^^^^
     |
help: a type alias with a similar name exists
     |
2172 -         ctx: *const IndexDecoderCtx,
2172 +         ctx: *const IndexDecoder,
     |
help: consider importing this union
    -->  inverted_index_bencher/src/ffi.rs:25:5
     |
25   +     use ffi::IndexDecoderCtx;
     |

error[E0412]: cannot find type `IndexDecoderCtx` in this scope
    --> /Users/glendc/code/github.com/redisearch/RediSearch/src/redisearch_rs/../../bin/redisearch_rs/optimised_test/build/inverted_index_bencher-3dd870ec90f6ca32/out/bindings.rs:2179:21
     |
1901 | / pub type IndexDecoder = ::std::option::Option<
1902 | |     unsafe extern "C" fn(
1903 | |         arg1: *mut IndexBlockReader,
1904 | |         arg2: *const IndexDecoderCtx,
1905 | |         out: *mut RSIndexResult,
1906 | |     ) -> bool,
1907 | | >;
     | |__- similarly named type alias `IndexDecoder` defined here
...
2179 |           ctx: *const IndexDecoderCtx,
     |                       ^^^^^^^^^^^^^^^
     |
help: a type alias with a similar name exists
     |
2179 -         ctx: *const IndexDecoderCtx,
2179 +         ctx: *const IndexDecoder,
     |
help: consider importing this union
    -->  inverted_index_bencher/src/ffi.rs:25:5
     |
25   +     use ffi::IndexDecoderCtx;
     |

error[E0412]: cannot find type `IndexDecoderCtx` in this scope
    --> /Users/glendc/code/github.com/redisearch/RediSearch/src/redisearch_rs/../../bin/redisearch_rs/optimised_test/build/inverted_index_bencher-3dd870ec90f6ca32/out/bindings.rs:2186:21
     |
1901 | / pub type IndexDecoder = ::std::option::Option<
1902 | |     unsafe extern "C" fn(
1903 | |         arg1: *mut IndexBlockReader,
1904 | |         arg2: *const IndexDecoderCtx,
1905 | |         out: *mut RSIndexResult,
1906 | |     ) -> bool,
1907 | | >;
     | |__- similarly named type alias `IndexDecoder` defined here
...
2186 |           ctx: *const IndexDecoderCtx,
     |                       ^^^^^^^^^^^^^^^
     |
help: a type alias with a similar name exists
     |
2186 -         ctx: *const IndexDecoderCtx,
2186 +         ctx: *const IndexDecoder,
     |
help: consider importing this union
    -->  inverted_index_bencher/src/ffi.rs:25:5
     |
25   +     use ffi::IndexDecoderCtx;
     |

error[E0412]: cannot find type `IndexDecoderCtx` in this scope
    --> /Users/glendc/code/github.com/redisearch/RediSearch/src/redisearch_rs/../../bin/redisearch_rs/optimised_test/build/inverted_index_bencher-3dd870ec90f6ca32/out/bindings.rs:2193:21
     |
1901 | / pub type IndexDecoder = ::std::option::Option<
1902 | |     unsafe extern "C" fn(
1903 | |         arg1: *mut IndexBlockReader,
1904 | |         arg2: *const IndexDecoderCtx,
1905 | |         out: *mut RSIndexResult,
1906 | |     ) -> bool,
1907 | | >;
     | |__- similarly named type alias `IndexDecoder` defined here
...
2193 |           ctx: *const IndexDecoderCtx,
     |                       ^^^^^^^^^^^^^^^
     |
help: a type alias with a similar name exists
     |
2193 -         ctx: *const IndexDecoderCtx,
2193 +         ctx: *const IndexDecoder,
     |
help: consider importing this union
    -->  inverted_index_bencher/src/ffi.rs:25:5
     |
25   +     use ffi::IndexDecoderCtx;
     |

error[E0412]: cannot find type `IndexDecoderCtx` in this scope
    --> /Users/glendc/code/github.com/redisearch/RediSearch/src/redisearch_rs/../../bin/redisearch_rs/optimised_test/build/inverted_index_bencher-3dd870ec90f6ca32/out/bindings.rs:2200:21
     |
1901 | / pub type IndexDecoder = ::std::option::Option<
1902 | |     unsafe extern "C" fn(
1903 | |         arg1: *mut IndexBlockReader,
1904 | |         arg2: *const IndexDecoderCtx,
1905 | |         out: *mut RSIndexResult,
1906 | |     ) -> bool,
1907 | | >;
     | |__- similarly named type alias `IndexDecoder` defined here
...
2200 |           ctx: *const IndexDecoderCtx,
     |                       ^^^^^^^^^^^^^^^
     |
help: a type alias with a similar name exists
     |
2200 -         ctx: *const IndexDecoderCtx,
2200 +         ctx: *const IndexDecoder,
     |
help: consider importing this union
    -->  inverted_index_bencher/src/ffi.rs:25:5
     |
25   +     use ffi::IndexDecoderCtx;
     |

error[E0412]: cannot find type `IndexDecoderCtx` in this scope
    --> /Users/glendc/code/github.com/redisearch/RediSearch/src/redisearch_rs/../../bin/redisearch_rs/optimised_test/build/inverted_index_bencher-3dd870ec90f6ca32/out/bindings.rs:2207:21
     |
1901 | / pub type IndexDecoder = ::std::option::Option<
1902 | |     unsafe extern "C" fn(
1903 | |         arg1: *mut IndexBlockReader,
1904 | |         arg2: *const IndexDecoderCtx,
1905 | |         out: *mut RSIndexResult,
1906 | |     ) -> bool,
1907 | | >;
     | |__- similarly named type alias `IndexDecoder` defined here
...
2207 |           ctx: *const IndexDecoderCtx,
     |                       ^^^^^^^^^^^^^^^
     |
help: a type alias with a similar name exists
     |
2207 -         ctx: *const IndexDecoderCtx,
2207 +         ctx: *const IndexDecoder,
     |
help: consider importing this union
    -->  inverted_index_bencher/src/ffi.rs:25:5
     |
25   +     use ffi::IndexDecoderCtx;
     |
 ```

--- a/build.sh
+++ b/build.sh
@@ -569,7 +569,6 @@ run_rust_tests() {
     RUST_TEST_OPTIONS="
       --doctests
       --codecov
-      $RUST_EXCLUDE_CRATES
       --ignore-filename-regex="varint_bencher/*,trie_bencher/*,inverted_index_bencher/*"
       --output-path=$BINROOT/rust_cov.info
     "
@@ -579,7 +578,7 @@ run_rust_tests() {
 
   # Run cargo test with the appropriate filter
   cd "$RUST_DIR"
-  RUSTFLAGS="${RUSTFLAGS:--D warnings}" cargo $RUST_EXTENSIONS test --profile=$RUST_PROFILE $RUST_TEST_OPTIONS --workspace $RUST_EXCLUDE_CRATES $TEST_FILTER -- --nocapture
+  RUSTFLAGS="${RUSTFLAGS:--D warnings}" cargo $RUST_EXTENSIONS test --profile=$RUST_PROFILE $RUST_TEST_OPTIONS --workspace $TEST_FILTER -- --nocapture
 
   # Check test results
   RUST_TEST_RESULT=$?
@@ -634,7 +633,7 @@ run_rust_valgrind_tests() {
         cargo $RUST_EXTENSIONS valgrind test \
         --profile=$RUST_PROFILE \
         $RUST_TEST_OPTIONS \
-        --workspace $RUST_EXCLUDE_CRATES $TEST_FILTER \
+        --workspace $TEST_FILTER \
         -- --nocapture
   fi
 

--- a/build.sh
+++ b/build.sh
@@ -548,6 +548,9 @@ run_rust_tests() {
 
   echo "Running Rust tests..."
 
+  # Tell Rust build scripts where to find the compiled static libraries
+  export BINDIR
+
   # Set Rust test environment
   RUST_DIR="$ROOT/src/redisearch_rs"
 

--- a/src/redisearch_rs/build_utils/src/lib.rs
+++ b/src/redisearch_rs/build_utils/src/lib.rs
@@ -94,30 +94,38 @@ pub fn rerun_cbinden(config: &cbindgen::Config) -> Result<(), Box<dyn std::error
 /// # Panics
 /// Panics if any required static library is not found in the expected location.
 pub fn link_static_libraries(libs: &[(&str, &str)]) {
-    let root = git_root().expect("Could not find git root for static library linking");
     let target_os = env::var("CARGO_CFG_TARGET_OS").unwrap_or_else(|_| "linux".to_string());
-    let target_arch = match env::var("CARGO_CFG_TARGET_ARCH").ok().as_deref() {
-        Some("x86_64") | None => "x64".to_owned(),
-        Some(a) => a.to_owned(),
-    };
 
-    // There are several symbols exposed by `libtrie.a` that we don't
-    // actually invoke (either directly or indirectly) in our benchmarks.
-    // We provide a definition for the ones we need (e.g. Redis' allocation functions),
+    // There may be several symbols exposed by the static library that we are trying to link
+    // that we don't actually invoke (either directly or indirectly) in our benchmarks.
+    // We will provide a definition for the ones we need (e.g. Redis' allocation functions),
     // but we don't want to be forced to add dummy definitions for the ones we don't rely on.
     // We prefer to fail at runtime if we try to use a symbol that's undefined.
-    // This is the default linker behaviour on macOS. On other platforms, the default
-    // configuration is stricter: it exits with an error if any symbol is undefined.
-    // We intentionally relax it here.
     if target_os == "macos" {
         println!("cargo:rustc-link-arg=-Wl,-undefined,dynamic_lookup");
     } else {
         println!("cargo:rustc-link-arg=-Wl,--unresolved-symbols=ignore-in-object-files");
     }
 
-    let bin_root = root.join(format!(
-        "bin/{target_os}-{target_arch}-release/search-community/"
-    ));
+    let bin_root = if let Ok(bin_root) = std::env::var("BINDIR") {
+        // The directory changes depending on a variety of factors: target architecture, target OS,
+        // optimization level, coverage, etc.
+        // We rely on the top-level build coordinator to give us the correct path, rather
+        // than duplicating the whole layout logic here.
+        PathBuf::from(bin_root)
+    } else {
+        // If one is not provided (e.g. `cargo` has been invoked directly), we look
+        // for a release build of the static library in the conventional location
+        // for the bin directory.
+        let root = git_root().expect("Could not find git root for static library linking");
+        let target_arch = match env::var("CARGO_CFG_TARGET_ARCH").ok().as_deref() {
+            Some("x86_64") | None => "x64".to_owned(),
+            Some(a) => a.to_owned(),
+        };
+        root.join(format!(
+            "bin/{target_os}-{target_arch}-release/search-community/"
+        ))
+    };
 
     for &(lib_subdir, lib_name) in libs {
         link_static_lib(&bin_root, lib_subdir, lib_name).unwrap();

--- a/src/redisearch_rs/build_utils/src/lib.rs
+++ b/src/redisearch_rs/build_utils/src/lib.rs
@@ -109,7 +109,9 @@ pub fn link_static_libraries(libs: &[(&str, &str)]) {
     // This is the default linker behaviour on macOS. On other platforms, the default
     // configuration is stricter: it exits with an error if any symbol is undefined.
     // We intentionally relax it here.
-    if target_os != "macos" {
+    if target_os == "macos" {
+        println!("cargo:rustc-link-arg=-Wl,-undefined,dynamic_lookup");
+    } else {
         println!("cargo:rustc-link-arg=-Wl,--unresolved-symbols=ignore-in-object-files");
     }
 

--- a/src/redisearch_rs/inverted_index_bencher/src/lib.rs
+++ b/src/redisearch_rs/inverted_index_bencher/src/lib.rs
@@ -15,6 +15,10 @@ pub mod ffi;
 redis_mock::bind_redis_alloc_symbols_to_mock_impl!();
 
 #[unsafe(no_mangle)]
+#[allow(non_upper_case_globals)]
+pub static mut RSGlobalConfig: *const c_void = std::ptr::null();
+
+#[unsafe(no_mangle)]
 pub extern "C" fn ResultMetrics_Free(metrics: *mut ::ffi::RSYieldableMetric) {
     if metrics.is_null() {
         return;

